Automatically create users and groups from sysusers.d(5) files

pmatilai opened the pull request:
This implements declarative user and group handling, ie #1032 and https://github.com/rpm-software-management/rpm/discussions/2277.

The executive summary: if you ship sysusers.d(5) files in your packages, you never need to muck with user or group creation in specs again. 

More details in the commits, but basically we now generate user and group requires and provides automatically, encode any sysusers data into the generated provides and autocreate these on install. A macro is provided for manually declaring users and groups from specs too.
The native tooling is systemd-sysusers, but it's possible to create a compatibility script that translates the sysusers syntax to other tooling (eg adduser/addgroup).

Dunno if all these make sense as separate commits for a final version, probably not, but should for discussion anyway.

pmatilai commented:
The last push added --replace handling (annoyingly complicated) and means to disable the entire install-time machinery (by commenting out the systemd-sysusers command macro)

pmatilai commented:
Documentation still needed for merging, but otherwise I think this is feature-complete now.

ffesti commented on macros.in:
```diff
@@ -56,6 +56,7 @@
 %__hg			@__HG@
 %__bzr			@__BZR@
 %__quilt		@__QUILT@
+%__systemd_sysusers	%{_bindir}/systemd-sysusers
```
This clearly needs a comment like "comment out to disable user and group creation"

pmatilai commented on lib/rpmscript.c:
```diff
@@ -645,6 +663,20 @@ rpmScript rpmScriptFromTriggerTag(Header h, rpmTagVal triggerTag,
     return script;
 }
 
+rpmScript rpmScriptFromArgv(Header h, ARGV_t argv, rpmscriptFlags flags, int chroot)
```
Oh, indeed. The whole rpmScriptFromArgv() is certainly just special purpose hack to begin with, but at least the tag should be an argument.

pmatilai commented:
Addressed the two review remarks, merged couple of commits to make more sense and perhaps most importantly, added logic to avoid '=' padding in the EVR strings where they really are not appropriate.

ffesti commented:
I wonder if in [b9c694e](https://github.com/rpm-software-management/rpm/pull/2432/commits/b9c694e5108bd6ae193484d01b7d59a8e94c4cd3) we should also check for the user and group of the installed files.

Other than that this looks all good!

pmatilai commented:
Rebased and tests for on-disk files verify added to all the test-cases in the PR.
We can't chown when running "make check" as a regular user but the CI runs as root (in a container, in case anybody wonders) so we just as well might: it passes --nouser --nogroup when running as regular user now so the tests dont need skipping as non-root.

pmatilai commented:
Since there appears to be no further feedback (thanks for the review @ffesti)...

martinpitt commented:
This helped a lot, but it's still applying the user creation and `%attr()` processing in the wrong order. I filed issue #3073 about it.
